src/btree/bt_read.c

```diff
@@ -125,17 +126,228 @@ __instantiate_birthmarks(WT_SESSION_IMPL *session, WT_REF *ref)
+    __wt_free(session, ref->page_las->birthmarks);
+
+err:
+    for (i = 0; i < ref->page_las->birthmarks_cnt; i++) {
+        birthmarkp = ref->page_las->birthmarks + i;
+        birthmarkp->instantiated = false;
+    }
+    WT_TRET(__wt_btcur_close(&cbt, true));
+    __wt_free(session, upd);
+
+    return (ret);
+}
+
+/*
+ * __las_page_instantiate --
+ *     Instantiate lookaside update records that are not on disk image in a recently read page.
+ */
+static int
+__las_page_instantiate(WT_SESSION_IMPL *session, WT_REF *ref)
+{
+    struct las_page_prepared_updates {
+        WT_ITEM key;
+        wt_timestamp_t timestamp;
+        uint64_t txnid;
+    } * las_preparep;
+    WT_BIRTHMARK_DETAILS *birthmarkp;
+    WT_CACHE *cache;
+    WT_CURSOR *cursor;
+    WT_CURSOR_BTREE cbt;
+    WT_DECL_ITEM(las_prepares);
+    WT_DECL_RET;
+    WT_ITEM las_key, las_value, next_las_key;
+    WT_PAGE *page;
+    WT_UPDATE *upd;
+    wt_timestamp_t durable_timestamp, las_timestamp;
+    size_t incr;
+    uint64_t birthmark_cnt, las_txnid, recno;
+    uint32_t i, las_btree_id, las_prepare_cnt, session_flags;
+    uint8_t prepare_state, upd_type;
+    const uint8_t *p;
+    int cmp, exact, instantiated_cnt;
+    bool birthmark_record, locked;
+
+    birthmarkp = NULL;
+    cursor = NULL;
+    page = ref->page;
+    upd = NULL;
+    birthmark_cnt = 0;
+    las_btree_id = S2BT(session)->id;
+    las_prepare_cnt = 0;
+    locked = false;
+    instantiated_cnt = 0;
+    birthmark_record = false;
+    recno = WT_RECNO_OOB;
+    session_flags = 0; /* [-Werror=maybe-uninitialized] */
+    WT_CLEAR(las_key);
+
+    cache = S2C(session)->cache;
+
+    if (ref->page_las->min_skipped_ts == WT_TS_MAX)
+        return __instantiate_birthmarks(session, ref);
+
+    __wt_btcur_init(session, &cbt);
+    __wt_btcur_open(&cbt);
+
+    WT_ERR(__wt_scr_alloc(session, 0, &las_prepares));
+    WT_STAT_CONN_INCR(session, cache_read_lookaside_skew_old);
+    WT_STAT_DATA_INCR(session, cache_read_lookaside_skew_old);
+    if (WT_SESSION_IS_CHECKPOINT(session))
+        WT_STAT_CONN_INCR(session, cache_read_lookaside_skew_old_checkpoint);
+
+    /* Open a lookaside table cursor. */
+    __wt_las_cursor(session, &cursor, &session_flags);
+
+    /*
+     * The lookaside records are in key and update order, that is, there will be a set of in-order
+     * updates for a key, then another set of in-order updates for a subsequent key. We find the
+     * most recent of the updates for a key and then insert that update into the page, then all the
+     * updates for the next key, and so on. In case if the birthmark record exists for that key,
+     * then insert birthmark record into the page.
+     */
+    __wt_readlock(session, &cache->las_sweepwalk_lock);
+    locked = true;
+    cursor->set_key(cursor, las_btree_id, &ref->page_las->min_las_key, WT_TS_MAX, WT_TXN_MAX);
+    for (; ret == 0; ret = cursor->next(cursor)) {
+        if (instantiated_cnt != 0) {
```

Comment:
instantiated_cnt can be replaced with a bool or it can go as a variable in the for loop.
Can explain the algorithm with a comment, it has become somewhat complicated now.

Reply:
Added another bool variable, still kept the instantiated_cnt variable, as this may be useful in debugging to find out how many keys are instantiated for the page.